lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -455,6 +474,49 @@ LogicalResult FirRegOp::canonicalize(FirRegOp op, PatternRewriter &rewriter) {
     return success();
   }
 
+  // For reset-less 1d array registers, replace an uninitialized element with
```

Comment:
This doesn't seem like a canonicalization.  It is choosing a value for initialization.